File: БР1.2/Москалец Данила/labs/lab1/express-typeorm-boilerplate-main/src/models/refresh-token.entity.ts
import {
    BaseEntity,
    Column,
    CreateDateColumn,
    Entity,
    ManyToOne,
    PrimaryGeneratedColumn,
} from 'typeorm';

import { User } from './user.entity';

@Entity()
export class RefreshToken extends BaseEntity {
    @PrimaryGeneratedColumn()
    id: number;

    @Column({ type: 'varchar', length: 1024, unique: true })
    token: string;

    @Column({ type: 'timestamp' })
    expiresAt: Date;

    @CreateDateColumn()
    createdAt: Date;

    @ManyToOne(() => User, (user) => user.refreshTokens, {
        nullable: false,
        onDelete: 'CASCADE',
    })
    user: User;
}

File: БР1.1/Захматов Юрий/labs/lab1/src/main/java/com/petproject/itmoacsbackend/auth/entities/RefreshTokenEntity.java
package com.petproject.itmoacsbackend.auth.entities;

import com.petproject.itmoacsbackend.users.entities.UserEntity;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "refresh_token")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class RefreshTokenEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "seq_refresh_token")
    @SequenceGenerator(
            sequenceName = "seq_refresh_token",
            name = "seq_refresh_token",
            allocationSize = 5
    )
    private Long id;

    @Column(nullable = false, unique = true)
    private String token;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private UserEntity user;

    @Column(nullable = false)
    private LocalDateTime expiresAt;

    @Builder.Default
    public Boolean revoked = false;

    public Boolean isExpired() {
        return LocalDateTime.now().isAfter(expiresAt);
    }
}
